rustls/src/lib.rs

```diff
@@ -520,6 +536,12 @@ mod sync {
     pub(crate) type Arc<T> = alloc::sync::Arc<T>;
 }
 
+/// Arc alias for this entire crate - may alias to either of these depending on the cfg used when building:
+/// - [`portable_atomic_util::Arc`](https://docs.rs/portable-atomic-util/latest/portable_atomic_util/struct.Arc.html)
+/// - [`alloc::sync::Arc`]
+#[allow(unused_qualifications)]
+pub type Arc<T> = crate::arc_alias::Arc<T>;
```

Comment:
maybe simplify like this & remove extra `mod arc_alias` indirection:
```suggestion
pub type Arc<T> = crate::sync::Arc<T>;
```

Reply:
When I tried `cargo doc` with this idea, it seems to reference `alloc::sync::Arc` from Rust nightly documentation. I am adding comments for now.